### PR TITLE
Label functions, note causes of bugs

### PR DESCRIPTION
**Pull request by TiKevin83**
Some of the functions in here are unlabeled despite having decent labels in pokecrystal (GSC reused yellow's options code).  Updated the most relevant ones.
This code has a bug/feature called "fast options" that reappears due to its reuse in GSC, added some explanation of it in comments

### CODE REVIEW

File: engine/menu/options.asm [hunk 1]
```diff
@@ -404,7 +404,7 @@ Func_41ee9:
 	ld [hl], "▶"
 	ret
 
-Func_41f06:
+InitOptions:
```
Comment (dannye) [inline engine/menu/options.asm:407]: This duplicates a label at https://github.com/pret/pokeyellow/blob/master/engine/menu/main_menu.asm#L125
Sorry for taking over a year to look at this.

Comment (TiKevin83) [reply engine/menu/options.asm:407]: Ah, thanks for taking a look at it dannye.  Do you have an alternative suggestion?

Comment (dannye) [reply engine/menu/options.asm:407]: The preexisting `InitOptions` is for initializing the player's option.
This routine appears to initialize the display of the options menu.
So perhaps `InitOptionsMenu`.
Be sure to run `make compare` to ensure the build passes.

File: docs/bugs_and_glitches.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+This bug (or feature!) results in all options being shifted left or right if the respective direction is pressed on the same frame the options menu is opened.
+The bug also exists in pokegold and pokecrystal.
+
+**Fix:** Update [engine/menu/options.asm](/engine/menu/options.asm)
+
+```diff
+  DisplayOptionMenu_:
++
++   call JoypadLowSensitivity
+    call InitOptions
```
Comment (dannye) [inline docs/bugs_and_glitches.md:32]: Sorry for being pedantic, but this now doesn't match the label in your PR.

**Comment by dannye** [comment]
Thanks!

**Pull request merged**